### PR TITLE
core: delete entity models

### PR DESCRIPTION
**Pull request by shramee**

### CODE REVIEW

File: crates/dojo-core/src/world.cairo [hunk 1]
```diff
@@ -492,11 +490,27 @@ mod world {
         /// * `model` - The name of the model to be deleted.
         /// * `query` - The query to be used to find the entity.
         fn delete_entity(ref self: ContractState, model: felt252, keys: Span<felt252>) {
```
Comment (tarrencev) [inline crates/dojo-core/src/world.cairo:492]: lets pass in the layout here similar to the other calls

Comment (shramee) [reply crates/dojo-core/src/world.cairo:492]: We get the layout from the model class at the moment, but we can add it as an arg.
Should we worry about the offset..?
At the moment, stuff added manually without `0` offset (not via set!) wouldn't be cleared probably.

Comment (tarrencev) [reply crates/dojo-core/src/world.cairo:492]: no we don't need the offset, we can clear the entire entity according to the layout

Comment (shramee) [reply crates/dojo-core/src/world.cairo:492]: Cool, I've hard coded that to `0` like we do for `set!`

Comment (shramee) [reply crates/dojo-core/src/world.cairo:492]: @tarrencev I've added #1197, let me know what you think!

Comment (tarrencev) [reply crates/dojo-core/src/world.cairo:492]: Responded there, let's pass the layout as an arg

**Review by tarrencev — CHANGES_REQUESTED** [review]

File: crates/dojo-core/src/world.cairo (hunk 1 above)
Comment (shramee) [reply crates/dojo-core/src/world.cairo]: Done 😉

**Comment by shramee** [comment]
Thanks for fixing contract address!

**Comment by tarrencev** [comment]
np, it is a bit annoying and we need to automate it

**Pull request merged**